tracer/src/Datadog.Trace/Util/Http/HttpRequestUtils.cs

```diff
@@ -0,0 +1,29 @@
+// <copyright file="HttpRequestUtils.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.Util.Http
+{
+    internal class HttpRequestUtils
+    {
+        private const string NoHostSpecified = "UNKNOWN_HOST";
+
+        internal static string GetUrl(string scheme, string host, string pathBase, string path, Func<string> getQueryString, ImmutableTracerSettings tracerSettings = null)
```

Comment:
I think you could improve performance slightly by changing the `getQueryString` in either one of two ways:
1. Change `Func<string>` to take an input object like `Func<object, string>`
2. Change it to `string` and pass the query string directly

The reason each of those would be an improvement is because the current delegate takes zero inputs and returns a string. Since it takes zero inputs, each time this method is called with a delegate, the delegate must capture the original request in the closure, which results in new delegates (i.e. objects) to be created for each call.

If you instead change the delegate to accept an object as input, you could pass the request as an argument so there's no additional state captured. This means you could re-use the same delegate over and over again but you just end up changing the input argument each time. Or, the other method, if you pass the query directly, then you'll likely just be retrieving the existing query string object from the request. Either way, it should reduce the number of allocations.